src/components/MerchantPayout/MerchantPayoutScreen.tsx

```diff
@@ -78,42 +82,52 @@ export const MerchantPayoutScreen: FunctionComponent<NavigationFocusInjectedProp
   const [isScanningEnabled, setIsScanningEnabled] = useState(true);
   const [vouchers, setVouchers] = useState<Voucher[]>([]);
   const [merchantCode, setMerchantCode] = useState("");
+  const [voucherStatus, setVoucherStatus] = useState<VoucherStatus>({
+    status: "VALID"
+  });
 
   useEffect(() => {
     if (isFocused) {
       setIsScanningEnabled(true);
     }
   }, [isFocused]);
 
+  const onCancel = (): void => {
+    setVoucherStatus({
+      status: "VALID"
+    });
+    setIsScanningEnabled(true);
+  };
+
   const onCheckVoucher = async (input: string): Promise<void> => {
     try {
       setIsScanningEnabled(false);
+      setVoucherStatus({ status: "CHECKING" });
       const serialArr = vouchers.map(voucher => voucher.serial);
       validateVoucherCode(input, serialArr);
       setVouchers([...vouchers, { serial: input, denomination: 2 }]);
       Vibration.vibrate(50);
-      setIsScanningEnabled(true);
+      //TODO: Send to API -> Valid, for now timeout
+      setTimeout(() => {
+        console.log("Sending to API");
+        setVoucherStatus({ status: "VALID" });
+        setIsScanningEnabled(true);
+      }, 1000);
     } catch (e) {
       setIsScanningEnabled(false);
-      Alert.alert(
-        "Error",
-        e.message || e,
-        [
-          {
-            text: "Dismiss",
-            onPress: () => setIsScanningEnabled(true)
-          }
-        ],
-        {
-          onDismiss: () => setIsScanningEnabled(true) // for android outside alert clicks
-        }
-      );
+      setVoucherStatus({ status: "INVALID", errorMessage: e.message });
     }
   };
 
   const onBarCodeScanned: BarCodeScannedCallback = event => {
     if (isFocused && isScanningEnabled && event.data) {
       onCheckVoucher(event.data);
+    } else {
+      setVoucherStatus({
+        status: "INVALID",
+        errorMessage: "Please try Scanning again or enter voucher manually",
+        errorTitle: "Error scanning"
+      });
```

Comment:
Any validation should be done within the `onCheckVoucher`

Reply:
Ohh, we weren't sure if we had to have an error scanning modal based on the figma design and the else statement was to reflect that. The `onCheckVoucher` is called only if something was scanned successfully (but not validated). But we can remove the else statement too if u want!
![Screenshot 2020-06-18 at 10 08 11 PM](https://user-images.githubusercontent.com/38589870/85031387-31241f00-b1b1-11ea-8e3a-1edc4daed239.png)

Reply:
ahh i see...currently our catch error will result in the first modal (Invalid status header) with the error message as either "Invalid voucher code" or "Duplicate voucher code". Would you like us to differentiate the errors and show the second modal (Error scanning header)?